juju/juju
# Iaas only commands

## Description of change

We need to gracefully reject IAAS only commands run on a CAAS model.
There's 2 commits here owing to the testing fallout caused by the change.
Commit 1: introduce a marker interface which is embedded in IAAS only commands. The base model command determines the model type and rejects IAAS only commands on CAAS models. The lookup of model type is done as early as possible, starting with during Init(). This allows the user to get feedback ASAP that the command isn;t suitable, even if they haven't supplied all required args etc This resulted in some refactoring of how client store etc was obtained. There was a large test fallout.
Commit 2: fix all the failing tests. The main requirement was to ensure a valid client store was available for all commands, even for tests which only called Init().

## QA steps

Run up both a CAAS and IAAS model. Ensure common commands work across both models, and IAAS only commands fail on the CAAS model. Use combinations of -m <model> and switch etc to test different model selection scenarios.
